### DISCUSSION TITLE
Using CodeQL on a .Net/C# Application

### DISCUSSION THREAD

**Discussion by glav** [community/community#25977] [Actions]
<p>I enabled CodeQL on an older project that uses .Net framework. Repo is <a href="https://github.com/glav/Glav.CacheAdapter" rel="nofollow noopener">here</a></p>
<p>Adding the default CodeQL workflow failed as the ‘Autobuild’ task failed. According to the docs, I should manually build the solution which is fine. I did that but then CodeQL analysis failed as it could not find any code to analyse. It is not clear where the compiled code or source should be located for this to work.</p>
<p>Any tips on that as it is unclear where to place the files to proceed?</p>
<p>The workflow that ran and failed that I mentioned is <a href="https://github.com/glav/Glav.CacheAdapter/runs/824975191?check_suite_focus=true" rel="nofollow noopener">here</a></p>
<p>cc: <a class="mention" href="/jhutchings1">@jhutchings1</a></p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/glav">@glav</a>,</p>
<p>I forked your repository and tested on my side, I can reproduce the same problems.</p>
<p>I have created an issue ticket (<a href="https://github.com/github/codeql-action/issues/98" rel="nofollow noopener"><strong>github/codeql-action#98</strong></a>) in the repository of the <strong>CodeQL Action</strong> to report the questions to the appropriate engineering team for further investigation and evaluation.</p>
<p>You can follow that issue ticket and add your comments on it.</p>

**Comment by jhutchings1** [comment, marked as answer]
<p>Engineering confirmed that this behavior is related to the <code>UseSharedCompilation</code> flag for msbuild and dotnet build. <a href="https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/troubleshooting-code-scanning#using-net-core-2-for-builds-in-linux" rel="nofollow noopener">https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/troubleshooting-code-scanning#using-net-core-2-for-builds-in-linux</a></p>
